File: packages/print-service-client/src/pos-kitchen-ticket.ts
/**
 * Comanda de cocina → agente KaiPrinters (`type: "pos-kitchen-ticket"`).
 * Ticket operativo (sin totales / sin gaveta).
 */

import type { PosSaleTicketCompany } from "./pos-sale-ticket";

export const POS_KITCHEN_TICKET_PAYLOAD_VERSION = 1;

export const POS_KITCHEN_TICKET_FOOTER_NOTE = "Comanda de cocina";

export type PosKitchenTicketLine = {
  name: string;
  quantity: number;
  notes?: string | null;
};

export type PosKitchenTicketPayload = {
  version: number;
  company: PosSaleTicketCompany;
  productionUnitName: string;
  fireNumber: number;
  accountLabel: string;
  tableCode?: string | null;
  branchName?: string | null;
  issuedAt: string;
  lines: PosKitchenTicketLine[];
  footerNote: string;
  /** true cuando es copia en POS/mesero (réplica). */
  isReplica?: boolean;
};

export type PosKitchenTicketPrintExtras = {
  filename: string;
  documentType?: string;
  internalFolio?: string;
  sourceApp?: string;
  purpose?: string;
  /** Alias de impresora en el agente (comanda UP / réplica). */
  printerDisplayLabel?: string | null;
};

File: kai-printers-desktop/src-tauri/src/pos_kitchen_ticket.rs
//! Comanda de cocina → JSON (`pos-kitchen-ticket`).

use crate::pos_sale_ticket_pdf::TicketCompany;
use anyhow::{Context, Result};
use serde::Deserialize;
use std::path::PathBuf;

#[derive(Debug, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct PosKitchenTicketLine {
    pub name: String,
    pub quantity: f64,
    pub notes: Option<String>,
}

#[derive(Debug, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct PosKitchenTicket {
    #[serde(default)]
    #[allow(dead_code)]
    pub version: i32,
    pub company: TicketCompany,
    pub production_unit_name: String,
    pub fire_number: i64,
    pub account_label: String,
    pub table_code: Option<String>,
    pub branch_name: Option<String>,
    pub issued_at: String,
    pub lines: Vec<PosKitchenTicketLine>,
    pub footer_note: String,
    #[serde(default)]
    pub is_replica: Option<bool>,
}

pub fn parse_pos_kitchen_ticket_from_value(
    value: &serde_json::Value,
) -> Result<PosKitchenTicket> {
    serde_json::from_value(value.clone()).context("parse pos-kitchen-ticket")
}

pub fn write_pos_kitchen_ticket_escpos_from_value(
    dir: &PathBuf,
    value: &serde_json::Value,
) -> Result<PathBuf> {
    crate::pos_kitchen_ticket_escpos::write_pos_kitchen_ticket_escpos_from_value(dir, value)
}
